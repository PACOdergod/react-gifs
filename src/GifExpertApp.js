
import React from 'react'
import { useState } from 'react'
import { AddSerie } from './components/AddSerie'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({ categoria=['perros'] }) => {

    const [series, setSeries] = useState( categoria )

    const handleAdd = ( serie )=>setSeries( state => [serie, ...state] )

    return (
        <>
            <h2> Gifs App </h2>
            <p>Cual tema te gusta?</p>
            <AddSerie handleAdd={ handleAdd }/>
            <hr></hr>


            <ol>
                {
                    series.map( serie => {
                        return <GifGrid categoria={ serie } key={serie}/>
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp
