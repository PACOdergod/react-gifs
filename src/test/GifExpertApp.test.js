import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import GifExpertApp from '../GifExpertApp';

describe('Probando <GifExpertApp />', () => {
    
    test('debe dibujarse bien', () => {
        
        const wrapper = shallow( <GifExpertApp/>)

        expect( wrapper ).toMatchSnapshot()
    })

    test('debe mostrar una lista de categoria', () => {
        
        const categories = ['memes', 'perros']
        const wrapper = shallow( <GifExpertApp categoria={ categories }/>)

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )

    })
    
    
})
