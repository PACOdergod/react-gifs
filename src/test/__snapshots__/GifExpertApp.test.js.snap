// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GifExpertApp /> debe dibujarse bien 1`] = `
<Fragment>
  <h2>
     Gifs App 
  </h2>
  <p>
    Cual tema te gusta?
  </p>
  <AddSerie
    handleAdd={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="perros"
      key="perros"
    />
  </ol>
</Fragment>
`;

exports[`Probando <GifExpertApp /> debe mostrar una lista de categoria 1`] = `
<Fragment>
  <h2>
     Gifs App 
  </h2>
  <p>
    Cual tema te gusta?
  </p>
  <AddSerie
    handleAdd={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="memes"
      key="memes"
    />
    <GifGrid
      categoria="perros"
      key="perros"
    />
  </ol>
</Fragment>
`;
